meson.build

```
@@ -381,11 +381,19 @@ luaold_interpreters = [
 
 lua_pc_name = ''
 
-if sys_windows or get_option('lua-interpreter') == 'lua'
-  if sys_windows and get_option('lua-interpreter') != 'lua'
-    warning('There is no "' + get_option('lua-interpreter')
-            + '" binding option for Windows, backing up to "lua"')
+if get_option('lua-interpreter') == 'luajit'
+  if sys_osx == true
+    # luajit on macro is broken, this means we need to generate our own dependency with our arguments, a library later still needs to link to luajit for the pagesize argument thingy
+    lua = declare_dependency(
+      include_directories: include_directories(lua.get_pkgconfig_variable('includedir')),
+      link_args: ['-L'+lua.get_pkgconfig_variable('libdir'),
+                  '-l'+lua.get_pkgconfig_variable('libname')]
+    )
+  else
+    lua = dependency('luajit', fallback : ['luajit', 'luajit_dep'])
```

This defines a `luajit` wrap, right? If so there should be a `luajit.warp` doesn't it?

---

Yeah, if luajit is not found in the system, it will fall back to the subproject called luajit (`subprojects/luajit.wrap`). If not available I guess it will show a warning. The lua_pc_name will propagate the 'luajit' name for the whole project.

Do you think it is not the desired behavior or maybe I missed something..?

---

There is no `luajit.wrap` in this PR. At least I can't find it.

---

Yeah, we can remove the fallback. It will be added with 'luajit' wrapper if that is the case.
```suggestion
    lua = dependency('luajit')
```

---

It would make more sense to me.